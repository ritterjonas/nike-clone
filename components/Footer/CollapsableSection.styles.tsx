import { colors, font, spacing } from '@/styles/variables';
import styled from 'styled-components';

export const FooterSection = styled.div`
  display: flex;
  flex-direction: column;
  gap: ${spacing.layout16};
`;

export const LinkTitle = styled.a`
  color: ${colors.neutral100};
  font-size: ${font.size16};
  font-weight: 700;
  line-height: 150%;
  text-transform: uppercase;
  text-decoration: none;
  padding-top: ${spacing.layout24};
`;

export const Collapsable = styled.div`
  padding: ${spacing.layout20} 0;
  border-bottom: 1px solid ${colors.neutral100};
  border-top: 1px solid ${colors.neutral100};
  cursor: pointer;
  display: flex;
  justify-content: space-between;
`;

export const CollapsableTitle = styled.div`
  color: ${colors.neutral100};
  font-size: ${font.size14};
  text-transform: uppercase;
`;

export const CollapsableIcon = styled.div`
  color: ${colors.neutral100};
  font-size: ${font.size32};
  font-weight: 100;
  line-height: 14px;
`;

export const CollapsableContent = styled.div`
  padding: 0 ${spacing.layout16};
`;
