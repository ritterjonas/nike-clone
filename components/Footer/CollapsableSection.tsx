import { useIsMobile } from '../../hooks/useIsMobile';
import {
  Collapsable,
  CollapsableContent,
  CollapsableIcon,
  CollapsableTitle,
  FooterSection,
  LinkTitle,
} from './CollapsableSection.styles';
import { useState } from 'react';

type CollapsableSectionProps = {
  title: string;
  children: React.ReactNode;
};

export default function CollapsableSection({
  children,
  title,
}: CollapsableSectionProps) {
  const [open, setOpen] = useState(false);
  const isMobile = useIsMobile();

  return (
    <FooterSection>
      {isMobile ? (
        <>
          <Collapsable onClick={() => setOpen(!open)}>
            <CollapsableTitle>{title}</CollapsableTitle>
            <CollapsableIcon>{open ? 'â€“' : '+'}</CollapsableIcon>
          </Collapsable>
          {open && <CollapsableContent>{children}</CollapsableContent>}
        </>
      ) : (
        <>
          <LinkTitle>{title}</LinkTitle>
          {children}
        </>
      )}
    </FooterSection>
  );
}
