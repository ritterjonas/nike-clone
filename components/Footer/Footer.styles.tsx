import { colors, font, spacing } from '@/styles/variables';
import styled from 'styled-components';

export const FooterContainer = styled.div`
  background-color: ${colors.neutral700};
  width: 100%;
  box-sizing: border-box;
  padding: 40px;
  padding-top: 16px;
  gap: ${spacing.layout16};
`;

export const FooterContent = styled.div`
  display: grid;
  grid-auto-columns: minmax(0, 1fr);
  grid-auto-flow: column;
  padding-bottom: 32px;
  border-bottom: 1px solid ${colors.neutral600};

  @media (max-width: 1023px) {
    grid-auto-flow: row;
  }
`;

export const FooterSection = styled.div`
  display: flex;
  flex-direction: column;
  gap: ${spacing.layout16};
  padding: 24px 0;
`;

export const FooterLinks = styled.div`
  display: flex;
  flex-direction: column;
  gap: ${spacing.layout8};
  margin-bottom: 24px;
`;

export const LinkTitle = styled.a`
  color: ${colors.neutral100};
  font-size: ${font.size16};
  font-weight: 700;
  line-height: 150%;
  text-transform: uppercase;
  text-decoration: none;
`;

export const Link = styled.a`
  color: ${colors.neutral200};
  font-size: ${font.size12};
  font-weight: 400;
  line-height: 150%;
  text-decoration: none;
`;

export const PaymentIcons = styled.div`
  display: flex;
  gap: ${spacing.layout4};
  flex-wrap: wrap;
`;

export const SocialIcons = styled.div`
  display: flex;
  gap: ${spacing.layout8};
`;

export const LicenseContainer = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding-top: 32px;

  @media (max-width: 1023px) {
    flex-direction: column;
    gap: ${spacing.layout32};
  }
`;

export const LicenseSection = styled.div`
  display: flex;
  gap: ${spacing.layout48};
`;

export const LicenseLink = styled.a`
  color: ${colors.neutral200};
  text-decoration: none;
  font-size: ${font.size12};
`;

export const LicenseText = styled.div`
  color: ${colors.neutral300};
  text-decoration: none;
  text-align: right;
  font-size: ${font.size12};
  line-height: 150%;

  @media (max-width: 1023px) {
    text-align: center;
  }
`;
