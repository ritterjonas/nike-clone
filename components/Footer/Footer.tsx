import Image from 'next/image';
import ReceiveNews from '../ReceiveNews/ReceiveNews';
import {
  FooterContainer,
  FooterContent,
  FooterLinks,
  FooterSection,
  LicenseContainer,
  LicenseSection,
  LicenseLink,
  LicenseText,
  Link,
  LinkTitle,
  PaymentIcons,
  SocialIcons,
} from './Footer.styles';
import CollapsableSection from './CollapsableSection';

export default function Footer() {
  const socialNames = ['facebook', 'instagram', 'youtube'];
  const paymentNames = [
    'Mastercard',
    'Visa',
    'Amex',
    'Elo',
    'Hipercard',
    'Discover',
    'Boleto',
  ];

  return (
    <footer data-testid='footer'>
      <ReceiveNews />
      <FooterContainer>
        <FooterContent>
          <FooterSection>
            <LinkTitle href='#'>Encontre Uma Loja Nike</LinkTitle>
            <LinkTitle href='#'>Cadastre-se para receber novidades</LinkTitle>
            <LinkTitle href='#'>Mapa do site</LinkTitle>
          </FooterSection>
          <CollapsableSection title='Ajuda'>
            <FooterLinks>
              <Link href='#'>Dúvidas Gerais</Link>
              <Link href='#'>Encontre seu Tamanho</Link>
              <Link href='#'>Entregas</Link>
              <Link href='#'>Pedidos</Link>
              <Link href='#'>Trocas e Devoluções</Link>
              <Link href='#'>Editar consentimento</Link>
              <Link href='#'>Preferências de Cookie</Link>
              <Link href='#'>Pagamentos</Link>
              <Link href='#'>Produtos</Link>
              <Link href='#'>Corporativo</Link>
              <Link href='#'>Fale Conosco</Link>
            </FooterLinks>
          </CollapsableSection>
          <CollapsableSection title='Sobre a nike'>
            <FooterLinks>
              <Link href='#'>Feito para Jogar</Link>
              <Link href='#'>Sustentabilidade</Link>
            </FooterLinks>
          </CollapsableSection>
          <FooterSection>
            <LinkTitle>Redes sociais</LinkTitle>

            <SocialIcons>
              {socialNames.map(name => (
                <Image
                  src={`/images/logo_redes_sociais_${name}.svg`}
                  alt={`${name} logo`}
                  width={32}
                  height={32}
                  key={name}
                />
              ))}
            </SocialIcons>

            <LinkTitle>Formas de pagamento</LinkTitle>

            <PaymentIcons>
              {paymentNames.map(name => (
                <Image
                  src={`/images/pgto_${name}.svg`}
                  alt={`${name} logo`}
                  width={48}
                  height={36}
                  key={name}
                />
              ))}
            </PaymentIcons>
          </FooterSection>
        </FooterContent>
        <LicenseContainer>
          <LicenseSection>
            <LicenseLink>Brasil</LicenseLink>
            <LicenseLink href='#'>Política de Privacidade</LicenseLink>
            <LicenseLink href='#'>Termos de Uso</LicenseLink>
          </LicenseSection>
          <LicenseSection>
            <LicenseText>
              <p>
                © 2021 Nike. Todos os direitos reservados. Fisia Comércio de
                Produtos Esportivos
              </p>
              <p>
                Ltda - CNPJ: 59.546.515/0016-10 Rua José Semião Rodrigues
                Agostinho, 1370 -
              </p>
              <p>
                Galpão Unidade 17 - Nível 0, 1 e 2 - CEP 06833-300 - Embu das
                Artes/SP.
              </p>
            </LicenseText>
          </LicenseSection>
        </LicenseContainer>
      </FooterContainer>
    </footer>
  );
}
