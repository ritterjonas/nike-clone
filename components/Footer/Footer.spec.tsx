import React from 'react';
import { render, screen } from '@testing-library/react';
import Footer from './Footer';

test('Footer renders without errors', () => {
  render(<Footer />);
  const footerElement = screen.getByTestId('footer');
  expect(footerElement).toBeInTheDocument();
});

test('Footer contains "ENCONTRE UMA LOJA NIKE"', () => {
  render(<Footer />);
  const footerTextElement = screen.getByText('ENCONTRE UMA LOJA NIKE');
  expect(footerTextElement).toBeInTheDocument();
});
