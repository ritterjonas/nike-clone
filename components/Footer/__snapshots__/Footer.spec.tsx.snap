// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> matches snapshot 1`] = `
<DocumentFragment>
  <footer
    data-testid="footer"
  >
    <div
      class="sc-65e1a52f-0 gSumcn"
    >
      <img
        alt="Nike logo"
        data-nimg="1"
        data-testid="receive-news-nike-logo"
        decoding="async"
        height="48"
        loading="lazy"
        src="/images/logo_nike.svg"
        style="color: transparent;"
        width="100"
      />
      <h3
        class="sc-65e1a52f-1 fvpAyM"
      >
        RECEBA AS NOVIDADES DA NIKE
      </h3>
      <h4
        class="sc-65e1a52f-2 hvsCYe"
      >
        Cadastre-se e seja um dos primeiros a saber de todas as novidades e ofertas.
      </h4>
      <button
        class="sc-c0409c3f-0 biZASl"
      >
        Cadastre-se
      </button>
    </div>
    <div
      class="sc-ee751fab-0 fKKYBL"
    >
      <div
        class="sc-ee751fab-1 iAVJcQ"
      >
        <div
          class="sc-ee751fab-2 dokCsw"
        >
          <a
            class="sc-ee751fab-4 dmmEvS"
            href="#"
          >
            Encontre Uma Loja Nike
          </a>
          <a
            class="sc-ee751fab-4 dmmEvS"
            href="#"
          >
            Cadastre-se para receber novidades
          </a>
          <a
            class="sc-ee751fab-4 dmmEvS"
            href="#"
          >
            Mapa do site
          </a>
        </div>
        <div
          class="sc-97d8eb7b-0 mANps"
        >
          <a
            class="sc-97d8eb7b-1 ejLDlf"
          >
            Ajuda
          </a>
          <div
            class="sc-ee751fab-3 dKatHz"
          >
            <a
              class="sc-ee751fab-5 cewPMy"
              href="#"
            >
              Dúvidas Gerais
            </a>
            <a
              class="sc-ee751fab-5 cewPMy"
              href="#"
            >
              Encontre seu Tamanho
            </a>
            <a
              class="sc-ee751fab-5 cewPMy"
              href="#"
            >
              Entregas
            </a>
            <a
              class="sc-ee751fab-5 cewPMy"
              href="#"
            >
              Pedidos
            </a>
            <a
              class="sc-ee751fab-5 cewPMy"
              href="#"
            >
              Trocas e Devoluções
            </a>
            <a
              class="sc-ee751fab-5 cewPMy"
              href="#"
            >
              Editar consentimento
            </a>
            <a
              class="sc-ee751fab-5 cewPMy"
              href="#"
            >
              Preferências de Cookie
            </a>
            <a
              class="sc-ee751fab-5 cewPMy"
              href="#"
            >
              Pagamentos
            </a>
            <a
              class="sc-ee751fab-5 cewPMy"
              href="#"
            >
              Produtos
            </a>
            <a
              class="sc-ee751fab-5 cewPMy"
              href="#"
            >
              Corporativo
            </a>
            <a
              class="sc-ee751fab-5 cewPMy"
              href="#"
            >
              Fale Conosco
            </a>
          </div>
        </div>
        <div
          class="sc-97d8eb7b-0 mANps"
        >
          <a
            class="sc-97d8eb7b-1 ejLDlf"
          >
            Sobre a nike
          </a>
          <div
            class="sc-ee751fab-3 dKatHz"
          >
            <a
              class="sc-ee751fab-5 cewPMy"
              href="#"
            >
              Feito para Jogar
            </a>
            <a
              class="sc-ee751fab-5 cewPMy"
              href="#"
            >
              Sustentabilidade
            </a>
          </div>
        </div>
        <div
          class="sc-ee751fab-2 dokCsw"
        >
          <a
            class="sc-ee751fab-4 dmmEvS"
          >
            Redes sociais
          </a>
          <div
            class="sc-ee751fab-7 kfyGjo"
          >
            <a
              href="#"
            >
              <img
                alt="facebook logo"
                data-nimg="1"
                decoding="async"
                height="32"
                loading="lazy"
                src="/images/logo_redes_sociais_facebook.svg"
                style="color: transparent;"
                width="32"
              />
            </a>
            <a
              href="#"
            >
              <img
                alt="instagram logo"
                data-nimg="1"
                decoding="async"
                height="32"
                loading="lazy"
                src="/images/logo_redes_sociais_instagram.svg"
                style="color: transparent;"
                width="32"
              />
            </a>
            <a
              href="#"
            >
              <img
                alt="youtube logo"
                data-nimg="1"
                decoding="async"
                height="32"
                loading="lazy"
                src="/images/logo_redes_sociais_youtube.svg"
                style="color: transparent;"
                width="32"
              />
            </a>
          </div>
          <a
            class="sc-ee751fab-4 dmmEvS"
          >
            Formas de pagamento
          </a>
          <div
            class="sc-ee751fab-6 eDSEcw"
          >
            <img
              alt="Mastercard logo"
              data-nimg="1"
              decoding="async"
              height="36"
              loading="lazy"
              src="/images/pgto_Mastercard.svg"
              style="color: transparent;"
              width="48"
            />
            <img
              alt="Visa logo"
              data-nimg="1"
              decoding="async"
              height="36"
              loading="lazy"
              src="/images/pgto_Visa.svg"
              style="color: transparent;"
              width="48"
            />
            <img
              alt="Amex logo"
              data-nimg="1"
              decoding="async"
              height="36"
              loading="lazy"
              src="/images/pgto_Amex.svg"
              style="color: transparent;"
              width="48"
            />
            <img
              alt="Elo logo"
              data-nimg="1"
              decoding="async"
              height="36"
              loading="lazy"
              src="/images/pgto_Elo.svg"
              style="color: transparent;"
              width="48"
            />
            <img
              alt="Hipercard logo"
              data-nimg="1"
              decoding="async"
              height="36"
              loading="lazy"
              src="/images/pgto_Hipercard.svg"
              style="color: transparent;"
              width="48"
            />
            <img
              alt="Discover logo"
              data-nimg="1"
              decoding="async"
              height="36"
              loading="lazy"
              src="/images/pgto_Discover.svg"
              style="color: transparent;"
              width="48"
            />
            <img
              alt="Boleto logo"
              data-nimg="1"
              decoding="async"
              height="36"
              loading="lazy"
              src="/images/pgto_Boleto.svg"
              style="color: transparent;"
              width="48"
            />
          </div>
        </div>
      </div>
      <div
        class="sc-ee751fab-8 kTieXV"
      >
        <div
          class="sc-ee751fab-9 eccrGi"
        >
          <a
            class="sc-ee751fab-10 hjIljV"
          >
            Brasil
          </a>
          <a
            class="sc-ee751fab-10 hjIljV"
            href="#"
          >
            Política de Privacidade
          </a>
          <a
            class="sc-ee751fab-10 hjIljV"
            href="#"
          >
            Termos de Uso
          </a>
        </div>
        <div
          class="sc-ee751fab-9 eccrGi"
        >
          <div
            class="sc-ee751fab-11 lksdLe"
          >
            <p>
              © 2021 Nike. Todos os direitos reservados. Fisia Comércio de Produtos Esportivos
            </p>
            <p>
              Ltda - CNPJ: 59.546.515/0016-10 Rua José Semião Rodrigues Agostinho, 1370 -
            </p>
            <p>
              Galpão Unidade 17 - Nível 0, 1 e 2 - CEP 06833-300 - Embu das Artes/SP.
            </p>
          </div>
        </div>
      </div>
    </div>
  </footer>
</DocumentFragment>
`;
